### PR TITLE
feat(open-api-page): redesign

### PR DESCRIPTION
**Pull request by thiskevinwang**
# Description

This updates the `open-api-page` component

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-kevin-boundary-api-docs-redesign-2-hashicorp.vercel.app/boundary/api-docs/auth-method-service) 🔎
- [Asana task](url) 🎟️
- https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12290%3A134875

## 🧪 Testing

- [ ] Visit the preview — https://dev-portal-git-kevin-boundary-api-docs-redesign-2-hashicorp.vercel.app/boundary/api-docs/auth-method-service
- [ ] Verify Back Link has been added and functions correctly
- [ ] Click around various service docs
- [ ] Verify the Breadcrumb links update and that final item receives darker color style
- [ ] Verify new `AccordionDisclosure` usage within api-docs body content

### CODE REVIEW

File: src/components/accordion-disclosure/accordion-disclosure.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+		// button renders
+		expect(getByRole('button')).toBeInTheDocument()
+		// title renders
+		expect(queryByText('A Title')).toBeInTheDocument()
+		// description renders
+		expect(queryByText('A Description')).toBeInTheDocument()
+	})
+
+	it('applies nested class to all nested AccordionDisclosure', () => {
```
Comment (BRKalow) [inline src/components/accordion-disclosure/accordion-disclosure.test.tsx:20]: tests! 👏

File: src/components/accordion-disclosure/types.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export interface AccordionDisclosureProps {
 	 * emphasis. Always shows regardless of the `AccordionDisclosure`'s
 	 * open/closed state.
 	 */
-	description?: string
+	description?: string | ReactNode
```
Comment (BRKalow) [inline src/components/accordion-disclosure/types.ts:16]: `string` is included in `ReactNode`:
```suggestion
	description?: ReactNode
```

File: src/components/accordion-disclosure/types.ts [hunk 3]
```diff
@@ -27,5 +27,5 @@ export interface AccordionDisclosureProps {
 	 * the disclosed content; labels like "Section 1" or "Section 2" is not very
 	 * descriptive.
 	 */
-	title: string
+	title: string | ReactNode
```
Comment (BRKalow) [inline src/components/accordion-disclosure/types.ts:30]: `string` is included in `ReactNode`:

```suggestion
	title: ReactNode
```

File: src/components/accordion-disclosure/accordion-disclosure.module.css [hunk 4]
```diff
@@ -1,97 +1,107 @@
...
+
+.nested {
+	box-shadow: var(--token-surface-base-box-shadow) !important;
+	& .root-expanded {
+		box-shadow: var(--token-surface-base-box-shadow) !important;
+	}
+	& .button {
+		box-shadow: none !important;
+	}
 }
```
Comment (ashleemboyer) [inline src/components/accordion-disclosure/accordion-disclosure.module.css:107]: Testing locally, I think this class works without the `!important`s. Would also appreciate some empty lines between the blocks. 🙏🏻

```suggestion
.nested {
	box-shadow: var(--token-surface-base-box-shadow);

	& .root-expanded {
		box-shadow: var(--token-surface-base-box-shadow);
	}

	& .button {
		box-shadow: none;
	}
}
```

File: src/components/open-api-page/index.tsx [hunk 5]
```diff
@@ -1,10 +1,9 @@
-import { useCallback, useState, useRef } from 'react'
+import { useRef } from 'react'
```
Comment (ashleemboyer) [inline src/components/open-api-page/index.tsx:1]: I'm seeing some unused code in here. Do you want to go ahead and remove?

- `setOperationState`
- `isExpanded`

File: src/pages/boundary/api-docs/[[...page]].tsx [hunk 6]
```diff
@@ -92,9 +106,19 @@ export async function getStaticProps({ params }) {
 	// Construct sidebar nav data levels
 	const sidebarNavDataLevels = [
 		{
+			backToLinkProps: { text: 'Boundary Documentation', href: '/boundary/' },
 			menuItems: [
 				{
-					heading: apiPageProps.info.title,
+					heading: 'API',
+				},
```
Comment (ashleemboyer) [inline src/pages/boundary/api-docs/[[...page]].tsx:113]: We want this to be the sidebar's `title`, not a menu item.

File: src/pages/boundary/api-docs/[[...page]].tsx [hunk 7]
```diff
@@ -92,9 +106,19 @@ export async function getStaticProps({ params }) {
...
+				{
+					title: 'Overview',
+					indexData: true,
+					path: '',
+					fullPath: '/boundary/api-docs/',
+				},
```
Comment (ashleemboyer) [inline src/pages/boundary/api-docs/[[...page]].tsx:119]: I'm not sure what `indexData` is here. Also, I think you only need either `path` or `fullPath`.

**Review by ashleemboyer — CHANGES_REQUESTED** [review]

File: src/pages/boundary/api-docs/[[...page]].tsx [hunk 8]
```diff
@@ -92,9 +106,19 @@ export async function getStaticProps({ params }) {
 	// Construct sidebar nav data levels
 	const sidebarNavDataLevels = [
 		{
+			backToLinkProps: { text: 'Boundary Documentation', href: '/boundary/' },
```
Comment (ashleemboyer) [inline src/pages/boundary/api-docs/[[...page]].tsx:109]: This text should actually be `Boundary Home` like other non /docs documentation pages have it. I see that's not what in the designs, but that change is deferred until the IA updates RFC is approved and in motion.

```suggestion
			backToLinkProps: { text: 'Boundary Home', href: '/boundary/' },
```

File: src/components/accordion-disclosure/accordion-disclosure.module.css (hunk 4 above)
Comment (thiskevinwang) [reply src/components/accordion-disclosure/accordion-disclosure.module.css:107]: Oops, I moved this entire block down specifically to not-need the `!important`, but forgot to remove them 😅

**Comment by ashleemboyer** [comment]
The mobile version of these pages aren't working yet; would you like to handle them in a separate PR?

File: src/components/open-api-page/index.tsx (hunk 5 above)
Comment (thiskevinwang) [reply src/components/open-api-page/index.tsx:1]: I believe I pushed a commit removing these in the middle of your review

Comment (ashleemboyer) [reply src/components/open-api-page/index.tsx:1]: 👍🏻

File: src/pages/boundary/api-docs/[[...page]].tsx (hunk 7 above)
Comment (thiskevinwang) [reply src/pages/boundary/api-docs/[[...page]].tsx:119]: Removing `indexData` and using only `fullPath`

**Comment by thiskevinwang** [comment]
> The mobile version of these pages aren't working yet; would you like to handle them in a separate PR?

@ashleemboyer sure! What exactly isn't working on mobile?
(I also don't have mobile specs for this)

**Comment by ashleemboyer** [comment]
> > The mobile version of these pages aren't working yet; would you like to handle them in a separate PR?
> 
> @ashleemboyer sure! What exactly isn't working on mobile? (I also don't have mobile specs for this)

@thiskevinwang the mobile versions of pages like this one have "levels" in the sidebar. I think what you'll need is near identical to how the sidebar works for /boundary/docs. Pretty sure just the content of the sidebar should differ.

**Review by ashleemboyer — APPROVED** [review]

**Pull request merged**